### DISCUSSION TITLE
Feature request: Dates and version counts

### DISCUSSION THREAD

**Discussion by losmurfs** [ben-manes/gradle-versions-plugin#572] [Ideas]
The purpose of this feature is to make it easier to spot libraries that are no longer being maintained and therefore might need to consider replacing and also spot libraries that are further behind than others and therefore may require prioritization as far as upgrading.  Here is an example of how the output with the new feature might look (json version):
```json
{
    "current": {
        "dependencies": [{
                "group": "com.fasterxml.jackson.core",
                "userReason": null,
                "version": "2.1.4",
                "released": "Feb 15, 2013",
                "versionsBehind": 86,
                "projectUrl": "http://wiki.fasterxml.com/JacksonHome",
                "name": "jackson-annotations"
            },
            ...
    "gradle": {
        "current": {
            "version": "7.3",
             "released": "Nov 15, 2021",
             "versionsBehind": 0,
            "reason": "",
            "isUpdateAvailable": true,
            "isFailure": false
        },
        ...
        "running": {
            "version": "6.8",
            "released": "Jan 08, 2021",
            "versionsBehind": 12,
            "reason": "",
            "isUpdateAvailable": false,
            "isFailure": false
        }
        ...
    "outdated": {
        "dependencies": [
            {
                "group": "com.braintreepayments.gateway",
                "available": {
                    "release": "3.13.0",
                    "released": "Nov 09, 2021",
                    "versionsBehind": 0,
                    "milestone": null,
                    "integration": null
                },
                "userReason": null,
                "version": "3.12.0",
                    "released": "Jul 15, 2021",
                    "versionsBehind": 1,
                "projectUrl": "https://www.braintreegateway.com",
                "name": "braintree-java"
            },
...
```

**Comment by ben-manes** [comment]
I believe this is the same as #398, but unfortunately we couldn't find a good way to capture that.
- We would want to not tie it to a repository like Maven Central, since dependencies can come from anywhere.
- The maven-metadata.xml includes a generic `lastUpdated` field ([example](https://repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/maven-metadata.xml)) but that does not tie to a version. A resolution strategy might reject a later version or a patch backported, so assuming it matches the report's version would be wrong.
- The file timestamp seems iffy because the repository might act as a cache rather than the originator (as JCenter did for Maven Central). 
- Maven's pom ([example](https://repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/3.0.4/caffeine-3.0.4.pom)) and Gradle's module ([example](https://repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/3.0.4/caffeine-3.0.4.module)) release descriptors don't include the date, either.

Unfortunately if it is not captured reliably during release then I don't think we can reliably infer it.

**Comment by losmurfs** [comment]
Thanks, but this feature would not be for concrete decision making, it
would be for making things stand out that might not otherwise, let's face
it: the date being in accurate would be an edge case, not requesting that
date drives logic, but instead date is reported as extra data for
decision making: If the latest version says it is from 2005, likely a dead
project or project has moved to a new groupId or artifactId has changed.
This info cannot be relied on but would help developers notice a neglected
library in a large project that they might not otherwise notice.

**Reply by ben-manes** [reply]
Perhaps a plugin dedicated to that goal would be a better option. You could work with these authors to see if all of your features could be met. /cc @mroeppis

https://github.com/mgm-tp/gradle-libactivity-plugin

**Comment by losmurfs** [comment]
Thanks
